unity/Assets/Scripts/AgentManager.cs

```diff
@@ -766,6 +766,44 @@ private void Update()
             payload.Add(new KeyValuePair<string, byte[]>("image", captureScreen()));
 		}
 	}
+	
+	private void resetImageSynthesis(Camera camera)
+	{
+		ImageSynthesis imageSynthesis = camera.gameObject.GetComponentInChildren<ImageSynthesis>();
+		if (imageSynthesis != null && imageSynthesis.enabled)
+		{
+			imageSynthesis.OnCameraChange();
+			imageSynthesis.OnSceneChange();
+		}
+	}
+
+	private void resetAllImageSynthesis()
+	{
+		foreach (var agent in this.agents)
+		{
+			resetImageSynthesis(agent.m_Camera);
+		}
+
+		foreach (var camera in this.thirdPartyCameras)
+		{
+			resetImageSynthesis(camera);
+		}
+	}
+	
+	public IEnumerator WaitOnResolutionChange(int width, int height) {
+		while (Screen.width != width || Screen.height != height) {
+			yield return null;
+		}
+		this.resetAllImageSynthesis();
+		this.primaryAgent.actionFinished(true);
+	}
+
+	public void ChangeResolution(float x, float y) {
+		int height = Convert.ToInt32(y);
+		int width = Convert.ToInt32(x);
+		Screen.SetResolution(width, height, false);
+		StartCoroutine(WaitOnResolutionChange(width, height));
+	}
```

Comment:
With the new action dispatcher, can we make this `ChangeResolution(int x, int y)`? I believe they were floats only because `ServerAction.x` and `ServerAction.y` are.

Reply:
Good catch. Changed.